nnfabrik/templates/transfer/transferred_trained_model.py

```diff
@@ -0,0 +1,292 @@
+            -> master
+            ---
+            transfer_data:            attach@{storage}
+            """.format(
+                storage=self._master.storage
+            )
+            return definition
+
+    class CollapsedHistory(dj.Part):
+        """
+        For the result of two or more transfer steps to be uniquely identifiable,
+        we compress its entire history (the keys of all previous steps) into a single hash (`collapsed_history`).
+        This table keeps track of this process and can be used to recursively retrieve the transfer history.
+        """
+
+        definition = """
+        next_collapsed_history:                 varchar(64)
+        -> master
+        """
+
+        @classmethod
+        def add_entry(cls, key):
+            key = {
+                k: key[k]
+                for k in [
+                    "model_fn",
+                    "model_hash",
+                    "dataset_fn",
+                    "dataset_hash",
+                    "trainer_fn",
+                    "trainer_hash",
+                    "seed",
+                    "data_transfer",
+                    "transfer_step",
+                    "collapsed_history",
+                ]
+            }
+            key["next_collapsed_history"] = make_hash(key)
+            cls.insert1(key, skip_duplicates=True)
+
+    def _transfer_recipe(self, transfer_step):
+        """
+        Combines multiple transfer recipes and their resitrictions as specified by post_restr attribute.
+        The combination is transfer-step-specific, meaning only the recipes relevant for a specific transfer step would be combined.
+
+        Combining recipes are pretty easy and the user does not need to interact with this method directly. Below is an example:
+        Let us assume you have two recipe tables: TrainerRecipe and ModelRecipe, the you can attach all these recipes to your
+        TransferTrainedModel table as follow:
+
+        ``` Python
+            TransferTrainedModel.transfer_recipe = [TrainerRecipe, ModelRecipe]
+        ```
+
+        The rest (combining the recipes and their restrictions) is taken care of by this method.
+
+        Args:
+            transfer_step (int): table population trasnfer step.
+
+        Returns:
+            string or datajoint AndList: A single or combined restriction of one or multiple recipes, respectively.
+        """
+
+        if isinstance(self.transfer_recipe, (list, tuple)):
+            # get the recipes that have an entry for a specific transfer step
+            transfer_recipe = []
+            for tr in self.transfer_recipe:
+                # check if an entry exists for a specific transfer step in the recipe
+                if tr & f"transfer_step = {transfer_step}":
+                    # if it exists add that entry to the list of recipes (relevant for a specific transfer step)
+                    transfer_recipe.append(tr & f"transfer_step = {transfer_step}")
```

Comment:
What happens if there is more than one recipe in the current recipe table at the current transfer step?

Reply:
They will all be appended and joined in an `AndList`.

Reply:
Ah, I see.